### PR TITLE
Standalone migration performance improvements

### PR DESCRIPTION
**Pull request by crisbeto**
Includes some performance improvements and fixes to the standalone migration that came up after testing it against a large internal project. The individual commits have more info, but here's a summary:

1. Significantly speeds up the same-file reference lookups that we have to do for the `bootstrapModule` migration. E.g. previously a single node lookup could take as long as 30s on a large app and now it takes ~30ms.
2. Avoids interrupting the entire migration if the language service throws an error when making a lookup.
3. Avoids having to re-traverse nodes that we've already traversed as a part of the `bootstrapModule` migration. This reduces the migration time even further, because we have to make fewer expensive language service lookups.

### CODE REVIEW

File: packages/core/schematics/test/standalone_migration_spec.ts [hunk 1]
```diff
@@ -3259,4 +3259,48 @@ describe('standalone migration', () => {
         export class AppComponent {}
       `));
      });
+
+  it('should be able to migrate a bootstrapModule call where the root component does not belong to the bootstrapped component',
```
Comment (crisbeto) [inline packages/core/schematics/test/standalone_migration_spec.ts:3263]: This test is unrelated to the changes in the PR. I just wanted to ensure that we handle this case since I saw something similar in the app I was testing.

**Review by devversion — APPROVED** [review]
